pkg/registry/resolver.go

```diff
@@ -147,6 +139,61 @@ func (r *registry) Resolve(name string, config api.MultiStageTestConfiguration)
 	return expandedFlow, nil
 }
 
+// mergeEnvironments joins two environment maps.
+// Elements in `dst` are overwritten by those in `src` if they target the same
+// variable.
+func mergeEnvironments(dst *api.TestEnvironment, src api.TestEnvironment) {
+	mergeMaps((*map[string]string)(dst), (map[string]string)(src))
+}
+
+// mergeDependencies joins to dependency maps.
+// Elements in `dst` are overwritten by those in `src` if they target the same
+// variable.
+func mergeDependencies(dst *api.TestDependencies, src api.TestDependencies) {
+	mergeMaps((*map[string]string)(dst), (map[string]string)(src))
+}
+
+func mergeMaps(dst *map[string]string, src map[string]string) {
+	if src == nil {
+		return
+	}
+	if *dst == nil {
+		*dst = make(api.TestEnvironment, len(src))
```

Comment:
Why does this assume a type for `dst`?

Reply:
Uh… thanks for catching it, but how is this legal?  Updated.

Reply:
OK, so only the pointer types are incompatible.  Updated again to remove the casts in the other `merge*` functions.